### ISSUE TITLE
AccountId defined twice

### ISSUE THREAD

**Issue by wackazong** [near/near-sdk-rs#780]
I noticed that `AccountId` is defined in `near_sdk::types` but also in `near_primitives_core::types`.

When using `near_sdk` version 4.0.0-pre.7 the `VMContext` uses the `AccountId` definition from `near_primitives_core::types`. On the other hand, e.g. `env::predecessor_account_id()` uses the definition from `near_sdk::types`.

Is that expected? To me it does not make much sense. I looked into the release histories of the packages and into the near-sdk-rs issue tracker but did not find any clues.

**Comment by miraclx** [comment]
Afaik, when the `near-sdk` `AccountId` was introduced, the `near-primitives` version was still a bit bulky to be used in smart contracts. It's been stripped down and refactored a lot since then, though. Maybe @austinabell, you can take a look.

**Comment by wackazong** [comment]
Ok, thanks for the explanation.

Thing is, it's being used in the definition of `VMContext`. If I do unit tests I do need to handle both versions. I use the `near-primitives` version globally now and convert from the old one using (for whom it may concern):

```
fn convert_old_account_id(old: OldAccountId) -> AccountId {
    AccountId::from_str(&old.to_string())
        .unwrap_or_else(|_| env::panic_str("INVALID_OLD_ACCOUNT_ID"))
}
```

**Comment by austinabell** [comment]
Ah, yes, so it's generally recommended to use `VMContextBuilder` to generate context. This type probably shouldn't be exposed from the SDK, I'll make a change for this soon

**Comment by austinabell** [comment]
Closed with #785
